index.js

```diff
@@ -18,10 +22,22 @@ class Chapter {
 
   map(array, callback) {
     const output = [];
-    for (let i = 0; i < array.length; i++) {
-      output.push(callback(array[i]));
+
+    if (typeof array === 'function') {
+      for (let i = 0; i < this.result.length; i++) {
+        output.push(array(this.result[i]));
+      }
+
+      this.result = output;
+
+      return this;
+    } else {
```

Comment:
since we are returning in the previous if - the `else` block is unnecessary here. In case if we go into `if`, this code won't be executed, since the function would already return.

Reply:
Removed `else` block